### PR TITLE
Add sudokusolver led display & add problem transformations

### PR DESCRIPTION
**Pull request by eloots**
- Visualize Sudoku solving activity on LEDs 9 and 10
  - Pulse LED 9 for ±15ms:
    - Green when sudoku problem can be handled immediately
    - Blue when problem has to be stashed because solver is busy
    - Red when problem is dropped due to stash buffer full
  - LED 10 is Green during the solving of a sudoku problem
    - Note that this allows to visualize HotSpot optimalisation;
      On a Raspberry Pi, the first sudoku takes ±1 second to solve.
      Once steady-state is reached, this drops to ±30 milliseconds.
- The LedStripDriver actor is instantiated separately so that it
  is accessible by the SudokuSolver actor.
- The SudokuProblemSender actor sends transformed versions of a
  single Sudoku problem. It renders it into different versions
  of the problem by rotating, mirroring, transposing, ... it.

### CODE REVIEW

File: common/src/main/scala/akkapi/cluster/LedStripDriver.scala [hunk 1]
```diff
@@ -58,33 +71,51 @@ class LedStripDriver private (context: ActorContext[LedStripDriver.Command],
...
+      run(hearbeatLedIsOn, weaklyUpLedIsOn = true, weaklyUpNodes, blinkers)
   }.receiveSignal {
     case (_, signal) if signal == PostStop =>
       resetAllLeds(strip)
       Behaviors.same
   }
 
+  private def createLedPulser(id: String): ActorRef[LedPulser.Command] = {
+    val blinker = context.spawnAnonymous(LedPulser(settings, context.self))
+    context.watchWith(blinker, BlinkerTerminated(id))
```
Comment (manuelbernhardt) [inline common/src/main/scala/akkapi/cluster/LedStripDriver.scala]: Maybe I'm missing something, but is `BlinkerTerminated` handled anywhere?

File: exercise_012_clustered_sudoku_solver/src/main/scala/akkapi/cluster/sudoku/package.scala [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    def remapNumbers(mapping: Map[Int, Int]): SudokuField = {
+      require(mapping.keySet == mapping.values.to(Set))
+      val mapping_0 = mapping + (0 -> 0)
+      SudokuField(sudokuField.sudoku.map { row =>
+        row.map(cell => Set(mapping_0(cell.head)))
+      })
+    }
+
+    def randomSwapAround: SudokuField = {
+      val swapAround = scala.util.Random.shuffle(Vector(1,2,3,4,5,6,7,8,9)).zipWithIndex.map{ case (a, b) => (a, b + 1)}.to(Map) + (0 -> 0)
```
Comment (manuelbernhardt) [inline exercise_012_clustered_sudoku_solver/src/main/scala/akkapi/cluster/sudoku/package.scala]: Nitpicking, but that's a long line :)

File: common/src/main/scala/akkapi/cluster/LedStripDriver.scala (hunk 1 above)
Comment (eloots) [reply common/src/main/scala/akkapi/cluster/LedStripDriver.scala]: Good catch. Adding the handling of this message.
More in general, this is part of a bigger context where a pulser could be stopped at some point in time (either autonomously based on some condition or controlled by the `LEDDriver` by sending some message). Right now, this isn't the case. Once started, a pulser stays alive until the application itself is stopped.

File: exercise_012_clustered_sudoku_solver/src/main/scala/akkapi/cluster/sudoku/package.scala (hunk 2 above)
Comment (eloots) [reply exercise_012_clustered_sudoku_solver/src/main/scala/akkapi/cluster/sudoku/package.scala]: Absolutely true! Pushed a simplification (I hope)

**Pull request merged**